pkg/cloud/services/compute/instance.go

```diff
@@ -244,7 +215,7 @@ func (s *Service) InstanceCreate(clusterName string, machine *clusterv1.Machine,
 		Name:             openStackMachine.Name,
 		ImageRef:         imageID,
 		FlavorName:       openStackMachine.Spec.Flavor,
-		AvailabilityZone: openStackMachine.Spec.AvailabilityZone,
+		AvailabilityZone: *machine.Spec.FailureDomain,
```

Comment:
As a precaution, you may want to guard against a potential nil pointer error here.

Reply:
Ah of course. Thx I'll fix that tomorrow

Reply:
Fixed